homeassistant/components/dsmr_reader/definitions.py

```diff
@@ -27,197 +29,202 @@ def tariff_transform(value):
     "dsmr/reading/electricity_delivered_1": {
         "name": "Low tariff usage",
         "icon": "mdi:flash",
```

Comment:
We don't set a custom icon if we set device class.

Reply:
Added the unit_of_measurement back in the definitions.py!

Reply:
That's good. Here my comment was about the icons though. We don't set a custom icon when setting a device class since in that case there's already an icon set by the frontend for the device class.

Reply:
Yes I see, but these icons are set earlier with, in my opinion, a good purpose;

With the icons set:
![icons set](https://tweakers.net/i/8NloROF7M35BRyBYkWh8mg9MwCk=/full-fit-in/4000x4000/filters:no_upscale():fill(white):strip_exif()/f/image/wHclvWjlZBgKVLLIux2nY4ZT.png?f=user_large)
With the icons removed from the definitions.py:
![icons not set](https://tweakers.net/i/Sgy4k3S_J7qsZY9fZQNm-uhnFEc=/full-fit-in/4000x4000/filters:no_upscale():fill(white):strip_exif()/f/image/Hy0K6d0M0jOXy02T20VHhZcg.png?f=user_large)

Of course these icons could be set in Lovelace, but I think it really helps the users by providing a corresponding icon.

I started this PR by adding the "Telegram timestamp", later on we saw no use of "device_classes" which helps HA tremendously by actually interpreting the timestamp as a timestamp, and as bonus the device_classess are added on all other sensors ;)

Let me know if you actually want to remove the icons from the definitions.py, as I think it helps the user a lot.

Reply:
Yes, please remove the icons.